RecoParticleFlow/PFClusterProducer/interface/PFRecHitQTests.h

```diff
@@ -381,6 +381,83 @@ class PFRecHitQTestHOThreshold : public PFRecHitQTestBase {
 };
 
 //
+//  Quality test that checks threshold as a function of ECAL eta-ring
+//
+#include "Calibration/Tools/interface/EcalRingCalibrationTools.h"
+#include "DataFormats/DetId/interface/DetId.h"
+#include "Geometry/CaloGeometry/interface/CaloGeometry.h"
+#include "Geometry/Records/interface/CaloGeometryRecord.h"
+class PFRecHitQTestECALMultiThreshold : public PFRecHitQTestBase {
+ public:
+  PFRecHitQTestECALMultiThreshold() {
+
+  }
+
+  PFRecHitQTestECALMultiThreshold(const edm::ParameterSet& iConfig):
+    PFRecHitQTestBase(iConfig)
+    {
+      thresholds_ = iConfig.getParameter<std::vector<double> >("thresholds");
+    }
+
+    void beginEvent(const edm::Event& event,const edm::EventSetup& iSetup) {
+      edm::ESHandle<CaloGeometry> pG;
+      iSetup.get<CaloGeometryRecord>().get(pG);
+      CaloSubdetectorGeometry const* endcapGeometry = pG->getSubdetectorGeometry(DetId::Ecal, EcalEndcap);
+      endcapGeometrySet_=false;
+      if (endcapGeometry) {
+        EcalRingCalibrationTools::setCaloGeometry(&(*pG));
+        endcapGeometrySet_=true;
+      }
+    }
+
+    bool test(reco::PFRecHit& hit,const EcalRecHit& rh,bool& clean,bool fullReadOut){
+      return fullReadOut || pass(hit);
+    }
+    bool test(reco::PFRecHit& hit,const HBHERecHit& rh,bool& clean){
+      return true;
+    }
+
+    bool test(reco::PFRecHit& hit,const HFRecHit& rh,bool& clean){
+      return true;
+    }
+    bool test(reco::PFRecHit& hit,const HORecHit& rh,bool& clean){
+      return true;
+    }
+
+    bool test(reco::PFRecHit& hit,const CaloTower& rh,bool& clean){
+      return true;
+    }
+
+    bool test(reco::PFRecHit& hit,const HGCRecHit& rh,bool& clean){
+      return true;
+    }
+
+ protected:
+    std::vector<double> thresholds_;
+    bool endcapGeometrySet_;
+
+    bool pass(const reco::PFRecHit& hit){
+      
+      // this is to skip endcap ZS for Phase2 until there is a defined geometry
+      // apply the loosest ZS threshold, for the first eta-ring in EE
+      if(!endcapGeometrySet_) {
+        // 0-169: EB eta-rings
+        // 170-208: EE- eta rings
+        // 209-247: EE+ eta rings
+        int firstEERing = 170;
```

Comment:
is this done correctly?
IIUC, this ad hoc cut on iring 170 is supposed to correspond to EE.
So, there should be a check that this is an EE hit, otherwise this branch is incorrectly applied to EB rechits with a wrong threshold.
This is seen e.g. in 21234 ttbar wf (D4) where ECAL clusters are all in the barrel (HGCAL is loaded separately)
![all_oldvsnew_ttbar14tev2023d4wf21234p0c_log10recopfclusters_particleflowclusterecal__reco_obj_energy](https://cloud.githubusercontent.com/assets/4676718/24394157/28021812-139b-11e7-94f1-2e5571b370b4.png)